src/lib/storage/disk_size.rb

```diff
@@ -130,6 +130,7 @@ def unit_multiplier(unit)
       #
 
       def +(other)
+        return DiskSize.unlimited if any_operand_unlimited?(other)
```

Comment:
You mean, now that I implemented that on my own without knowing that the concept exists in Ruby? ;-)

Not sure if everything would work well with the standard Ruby Fixnum I am using here for size_k in the DiskSize when I add Float::INFINITY to the mix. It might or might not work as expected. There might be weird side effects. My implementation is now well tested - the test cases should cover all scenarios IMHO.